docs/guides/publish.md

```diff
@@ -13,6 +13,19 @@ build it by default. This means that your project may not be ready for distribut
 the effect of declaring a build system in the
 [project concept](../concepts/projects.md#build-systems) documentation.
 
+!!! note
+
+    To avoid accidentally publishing an internal package to PyPI, set this classifier that is valid
+    but gets rejected by PyPI:
```

Comment:
What happens with other package indexes?

Reply:
They should allow the upload, but in general this is undefined, as in the OP:

> Note that we cannot handle e.g. a GitLab repository that is set to public accidentally. It falls into the domain of registry vendors to have guardrails (e.g. private-by-default registries when the status of the source is unknown)

Reply:
We should probably say that too then.

Reply:
I mean, sure, but isn't saying that you need to read the manual to your registry to operate it securely both evident and outside of the scope of uv? I don't think anyone reads the section as keeping the package private when you switch your GitLab registry setting to public. (And the registry I checked start with private by default at least for private repos and have extra steps before switching it to public).

Reply:
I don't think we need to talk about other indexes in depth here, but if we're going to recommend something we need to be clear about the caveats Just asking for something like "When not publishing to PyPI, the classifier may or may not be respected."